alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDocLevelMonitorFanOutAction.kt

```diff
@@ -575,6 +575,11 @@ class TransportDocLevelMonitorFanOutAction
 
             // Before the "|" is the doc id and after the "|" is the index
             val docIndex = it.key.split("|")
+            val additionalFields = this.fetchDocForFinding(
```

Comment:
Making a search call per doc is not going to scale.

We have the document in memory as part of the monitor execution. Can we add these additional fields when we fetch that document and then carry them forward from memory?

Reply:
@engechas findings are sparse
IMO we should do a second search in bulk for all finding-generating docs